metrics/measured_http/http.go

```diff
@@ -62,11 +62,37 @@ func (h *MeasuredHandler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 	begin := h.clk.Now()
 	rwws := &responseWriterWithStatus{w, 0}
 
+	// Use the method string only if it's a recognized HTTP method. This avoids
+	// ballooning timeseries with invalid methods from public input.
+	var method string
+	switch r.Method {
+	case http.MethodGet:
+		fallthrough
+	case http.MethodHead:
+		fallthrough
+	case http.MethodPost:
+		fallthrough
+	case http.MethodPut:
+		fallthrough
+	case http.MethodPatch:
+		fallthrough
+	case http.MethodDelete:
+		fallthrough
+	case http.MethodConnect:
+		fallthrough
+	case http.MethodOptions:
+		fallthrough
+	case http.MethodTrace:
+		method = r.Method
+	default:
+		method = "unknown"
```

Comment:
Is an unknown method unlikely to ever get this far? Akamai definitely doesn't allow these requests through and as far as I can tell nginx will 405 requests with unknown methods.